### PR TITLE
Maintenance: replace most NULL with nullptr

### PR DESCRIPTION
**Pull request by yadij**

### CODE REVIEW

File: src/acl/external/eDirectory_userip/ext_edirectory_userip_acl.cc [hunk 1]
```diff
@@ -1722,7 +1722,7 @@ MainSafe(int argc, char **argv)
                     local_printfx("BH message=\"(SearchFilterLDAP: %s)\"\n", ErrLDAP(x));
                 } else {
                     edui_ldap.err = -1;
-                    debug("SearchFilterLDAP(-, NULL) -> Length: %u\n", x);
+                    debug("SearchFilterLDAP(-, nullptr) -> Length: %u\n", x);
```
Comment (rousskov) [inline src/acl/external/eDirectory_userip/ext_edirectory_userip_acl.cc:1725]: This code change alters helper output. Please undo if this was not intentional. I am not requesting any changes here, and am not implying that this is the only functionality change in this PR.

**Review by rousskov — APPROVED** [review]
```
$ git grep -w 'NULL' upstream/master  '*.cc' | wc -l
711

$ git grep -w 'NULL' yadij/cleanup-nullptr-auto  '*.cc' | wc -l
345
```

Thank you for removing many of the remaining C++ NULLs.

```
$ git grep '[!=]=.NULL' upstream/master  '*.cc' | wc -l
153

$ git grep '[!=]=.NULL' yadij/cleanup-nullptr-auto '*.cc' | wc -l
153
```

This PRs leaves dozens (and probably hundreds) of NULLs in C++ code, including some common trivial cases. I would very much prefer to see no NULLs in C++ code after this PR. If you can safely improve your script to cover more cases, please do so, but I do not insist on this PR fully addressing the NULL problem in Squid.

Disclaimer: I have not reviewed most of the low-level changes in this PR.

File: src/acl/external/eDirectory_userip/ext_edirectory_userip_acl.cc (hunk 1 above)
Comment (yadij) [reply src/acl/external/eDirectory_userip/ext_edirectory_userip_acl.cc:1725]: This was intentional. The code is nullptr not ```NULL/0``` nowdays, so fixing the debug message is useful.

**Comment by yadij** [comment]
> ```
> $ git grep '[!=]=.NULL' yadij/cleanup-nullptr-auto '*.cc' | wc -l
> 153
> ```

These remainders are all boolean conditions involving NULL comparison. How to replace these led to controversy over what was correct coding style previously, so I am leaving out of this PR in order to facilitate easy merge.

**Comment by rousskov** [comment]
>  yadij added `M-cleared-for-merge` and removed `S-waiting-for-author` `S-waiting-for-reviewer` labels

@yadij, please note that, from GitHub and Anubis point of view, this PR is still waiting for @kinkie review.

**Comment by rousskov** [comment]
For the corresponding v6 change, see #1433.

**Comment by yadij** [comment]
backported to v6

**Pull request closed without merging**